### DISCUSSION TITLE
How to resolve deprecation warning about decorators/route handlers

### DISCUSSION THREAD

**Discussion by bunny-therapist** [litestar-org/litestar#3551] [Q&A] [answered]
Using any of the route handler decorators `get`, `post`, etc now causes the warning `"Semantic HTTP route handler classes are deprecated and will be replaced by functional decorators in Litestar 3.0.`

How do I resolve this? I checked the [docs](https://docs.litestar.dev/2/usage/routing/overview.html#registering-routes), but they still tell you to use these decorators which cause the warning.

The only way I can see right now is to directly use `HTTPRouteHandler` and pass in the method, e.g. `HttpMethod.GET`, but that does not sound to me like a "functional decorator"?

What is the action I am supposed to take based on this deprecation warning?

**Comment by provinzkraut** [comment, marked as answer]
Hey @bunny-therapist, thanks for reporting this! It seems like this is an overzealous warning, you're not doing everything wrong here.

It is this code which produces the warning: 
https://github.com/litestar-org/litestar/blob/84f51c8afc3203cd4914922b2ec3c1e92d5d40ba/litestar/handlers/http_handlers/decorators.py#L49

It was intended to warn if you subclass one of the decorators, but we seem to have forgotten to add an exception for the handler classes themselves. Would you mind opening a proper issue for this so we can track and fix it?

**Comment by bunny-therapist** [comment]
Thanks. Will do.

**Reply by bunny-therapist** [reply]
https://github.com/litestar-org/litestar/issues/3552
